### PR TITLE
feat(dbt): deprecate old style dbt project configuration for Dagster

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
Deprecate the following:

- `dagster_auto_materialize_policy`
- `dagster_freshness_policy`

## How I Tested These Changes
N/A

### CODE REVIEW

**Comment by rexledesma** [comment]
Current dependencies on/for this PR:
* master
  * **PR #15324** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15324" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15324?utm_source=stack-comment).

**Review by sryza — COMMENTED** [review]
The auto-materialize and freshness changes look good. I think the group changes are no longer applicable.

File: python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py [hunk 1]
```diff
@@ -113,8 +114,15 @@ def default_freshness_policy_fn(node_info: Mapping[str, Any]) -> Optional[Freshn
     if freshness_policy:
         return freshness_policy
 
-    # TODO: Remove this legacy configuration in 0.20.0
     legacy_freshness_policy_config = node_info["config"].get("dagster_freshness_policy", {})
+    deprecation_warning(
```
Comment (OwenKephart) [inline python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py:118]: doesn't this end up emitting warnings for people who have no freshness policy set at all? I think we should do this like:

```python
legacy_freshness_policy = _legacy_freshness_policy_fn(legacy_freshness_policy_config)
if legacy_freshness_policy:
    # deprecation warning
return legacy_freshness_policy
```

**Review by OwenKephart — CHANGES_REQUESTED** [review]
minor comment

**Review by OwenKephart — APPROVED** [review]

**Pull request merged**